openwisp_controller/config/api/serializers.py

```diff
@@ -277,6 +286,8 @@ def update(self, instance, validated_data):
 
 class DeviceGroupSerializer(BaseSerializer):
     meta_data = serializers.JSONField(required=False, initial={})
+    templates = FilterTemplatesByOrganization(many=True)
```

Comment:
I guess that here you will have to prepare a subclass of this: `FilterGroupTemplates`, eg:

```python
# warning: untested code wrote from the top of my head!

class FilterGroupTemplates(FilterTemplatesByOrganization):
    def get_queryset(self):
        return super().get_queryset().exclude(Q(default=True) | Q(required=True))
```